File: ios-app/LaVeinteDigital/IMSS/Vault/ImssCredentialModels.swift
import Foundation

/// Modelos de credenciales IMSS (port de `ImssCredentialModels.kt`).
enum ImssPortal: String, CaseIterable, Codable, Hashable {
    case tuPerfil = "tuperfil"
    case tarjetonDigital = "tarjetondigital"

    var displayName: String {
        switch self {
        case .tuPerfil: return "Tu Perfil IMSS"
        case .tarjetonDigital: return "Tarjetón Digital IMSS"
        }
    }

    var host: String {
        switch self {
        case .tuPerfil: return "tuperfil.imss.gob.mx"
        case .tarjetonDigital: return "rh.imss.gob.mx"
        }
    }

    var sourceValue: String {
        switch self {
        case .tuPerfil: return "TU_PERFIL"
        case .tarjetonDigital: return "TARJETON_DIGITAL"
        }
    }

    var loginURL: String {
        switch self {
        case .tuPerfil: return "https://tuperfil.imss.gob.mx/guitpei-web/login"
        case .tarjetonDigital: return "https://rh.imss.gob.mx/Personal/TarjetonDigital/"
        }
    }
}

/// Payload de credenciales cifrado en la bóveda.
/// Los nombres de campo coinciden con el JSON del shell Android.
struct ImssCredentialPayload: Codable, Equatable {
    var username: String
    var password: String
    var credentialVersion: Int = 1
    /// Tarjetón Digital: valor real del `<option>` de Delegación (ej. "17").
    var delegacionValue: String?
    /// Tarjetón Digital: nombre visible de la delegación (ej. "MICHOACAN").
    var delegacionLabel: String?
}

File: android-app/app/src/main/java/com/laveintedigital/app/imss/credentials/ImssCredentialModels.kt
package com.laveintedigital.app.imss.credentials

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Metadata for an IMSS credential vault entry.
 * The actual ciphertext and IV are stored in DataStore, not Room.
 */
@Entity(tableName = "imss_credentials")
data class ImssCredentialEntity(
    @PrimaryKey val portalId: String,  // "tuperfil" or "tarjetondigital"
    val credentialVersion: Int = 1,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
)

enum class ImssPortal(val id: String, val displayName: String, val host: String) {
    TU_PERFIL("tuperfil", "Tu Perfil IMSS", "tuperfil.imss.gob.mx"),
    TARJETON_DIGITAL("tarjetondigital", "Tarjetón Digital IMSS", "rh.imss.gob.mx"),
}

data class ImssCredentialPayload(
    val username: String,
    val password: String,
    val credentialVersion: Int = 1,
    /** Tarjetón Digital: valor real del `<option>` de Delegación (ej. "17"). */
    val delegacionValue: String? = null,
    /** Tarjetón Digital: nombre visible de la delegación (ej. "MICHOACAN"). */
    val delegacionLabel: String? = null,
)
